boozallen/aissemble
# #176 Remove migration utils in prep for incroporation into Baton

This ticket will focus on moving an initial utils class from aiSSEMBLE from com.boozallen.aissemble.upgrade to Baton in order to set the pattern for the remaining classes *FileUtils.java* will be migrated from aiSSEMBLE to Baton
 * Migrate utility in the Baton repo ✅
   * Remove mentions of aiSSEMBLE if present ✅
 * Identify and write unit tests for the relevant util methods✅ 
   * replaceLiteralInFile✅ 
   * replaceInFile ✅
   * modifyRegexMatchInFile ✅
   * getRegExCaptureGroups ✅
   * hasRegExMatch ✅
 * Verify the migrated utility works in Baton ✅
   * Update test steps and run through them ✅ 
 * Update the documentation for Baton to illustrate the new capability ✅
   * Update javadoc description if needed ✅
   * Add an example project in baton directory that uses utils to do a simple migration ✅
 * Nice to have: move MigrationUtils.java to Baton (It's super bare-bones and simple) ✅

NOTE: The following classes will not have unit tests written for them due to their simplicity
 * readAllFileLines
 * writeFile
 * getIndent
 * indentValues

 1. Pull down and open Baton
`git clone git@github.org:TechnologyBrewery/baton`
 2. Confirm the Baton bootstrap build is successful and tests pass
`mvn clean install -Pbootstrap -Dmaven.build.cache.enabled=false`
 3. Confirm the Baton example projects build successfully
`mvn clean install -Pdefault -Dmaven.build.cache.enabled=false`
 4. Pull down and open aiSSEMBLE
`git clone git@github.org:boozallen/aissemble`
 5. Checkout the branch for this ticket
`git checkout #176`
 6. Change the version of baton in the build parent pom to "1.1.0-SNAPSHOT"
 7. Run the baton migration unit tests successfully
`./mvnw clean install -pl :foundation-upgrade -amd -Dmaven.build.cache.enabled=false`
